File: examples/agent-framework-webhooks/typescript/src/adapters/anthropic.ts
import type { InboundEmail } from "@e2a/sdk/v1";

import type { ReplyAgent } from "../contracts.js";
import { REPLY_INSTRUCTIONS, emailPrompt } from "../prompt.js";

export interface AnthropicResult {
  content: readonly unknown[];
}

export type AnthropicRun = (prompt: string) => Promise<AnthropicResult>;

function textBlock(block: unknown): string | undefined {
  if (!block || typeof block !== "object") return undefined;
  const value = block as { type?: unknown; text?: unknown };
  return value.type === "text" && typeof value.text === "string" ? value.text : undefined;
}

export class AnthropicReplyAgent implements ReplyAgent {
  constructor(private readonly run: AnthropicRun) {}

  async reply(email: InboundEmail): Promise<string> {
    const result = await this.run(emailPrompt(email));
    return result.content.flatMap((block) => {
      const text = textBlock(block);
      return text === undefined ? [] : [text];
    }).join("\n");
  }
}

interface AnthropicClient {
  messages: {
    create(input: {
      model: string;
      max_tokens: number;
      system: string;
      messages: Array<{ role: "user"; content: string }>;
    }): Promise<AnthropicResult>;
  };
}

export interface AnthropicSDK {
  Anthropic: new () => AnthropicClient;
}

/** Build the production adapter from the official `@anthropic-ai/sdk` export. */
export function createAnthropicReplyAgent(
  sdk: AnthropicSDK,
  env: Record<string, string | undefined> = process.env,
): AnthropicReplyAgent {
  const client = new sdk.Anthropic();
  return new AnthropicReplyAgent((prompt) => client.messages.create({
    model: env.ANTHROPIC_MODEL ?? "claude-opus-4-8",
    max_tokens: 1024,
    system: REPLY_INSTRUCTIONS,
    messages: [{ role: "user", content: prompt }],
  }));
}

File: examples/agent-framework-webhooks/python/agent_webhooks/adapters/anthropic.py
"""Anthropic Messages SDK reply adapter."""

import os
from collections.abc import Awaitable, Callable
from typing import Any

from e2a import AsyncInboundEmail

from agent_webhooks.prompt import REPLY_INSTRUCTIONS, email_prompt

AnthropicRun = Callable[[str], Awaitable[Any]]
AnthropicClose = Callable[[], Awaitable[None]]


class AnthropicReplyAgent:
    def __init__(
        self, run: AnthropicRun, close: AnthropicClose | None = None
    ) -> None:
        self._run = run
        self._close = close

    @classmethod
    def from_env(cls) -> "AnthropicReplyAgent":
        from anthropic import AsyncAnthropic

        client = AsyncAnthropic()
        return cls(
            lambda prompt: client.messages.create(
                model=os.getenv("ANTHROPIC_MODEL", "claude-opus-4-8"),
                max_tokens=1024,
                system=REPLY_INSTRUCTIONS,
                messages=[{"role": "user", "content": prompt}],
            ),
            close=client.close,
        )

    async def aclose(self) -> None:
        if self._close is not None:
            await self._close()

    async def reply(
        self, email: AsyncInboundEmail, conversation_id: str
    ) -> str:
        del conversation_id
        result = await self._run(email_prompt(email))
        return "\n".join(
            block.text
            for block in result.content
            if getattr(block, "type", None) == "text"
            and isinstance(getattr(block, "text", None), str)
        )
